pkg/lib/synthesize/generate.go

```diff
@@ -0,0 +1,143 @@
+package synthesize
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
+	serializer "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/kubectl/pkg/util/slice"
+)
+
+var (
+	CsvAlreadyExistInBundleError = errors.New("a ClusterServiceVersion already exist in the bundle")
+	NoCsvInferranceFileFoundError = errors.New("no file that contains the non-inferable CSV information is found")
```

Comment:
Sp: only one R in infererence

Also, this error message makes sense to us but I worry that it’s not clear to newer users.

Reply:
I think that is fine. If a user inputs a bundle with no CSV or non-inferable CSV info, the error would be `fmt.Errorf("a bundle should contain either a CSV or non-inferable CSV information in the %s directory", s.metadataDir)`